### PR TITLE
fix: logging interface are never exported

### LINKED ISSUE #189: Add logging documentation
#### Reason/Context
I forgot to add logging documentation when I implemented the logging interface in #170

#### Description
I would add a single document explaining the logging interface the library uses.

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR started with being a simple doc PR, but apparently, the logging interface is never exposed and unusable...

This PR fixes this by exporting the type and adapting tests to use the exposed type to make sure it stays like that.

**Related issue(s)**
Fixes #189

### CODE REVIEW

File: docs/logging.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Logging
+When you are generating models this library uses a detached logging module so you can integrate your own logging implementation based on your needs. By default nothing is logged to console or otherwise.
+
+The library uses 4 different logging levels.
```
Comment (magicmatatjahu) [inline docs/logging.md:4]:
```suggestion
The library uses 4 different logging levels:
```

File: docs/logging.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import {ModelLoggingInterface, Logger} from '@asyncapi/generator-model-sdk'; 
+const customLogger: ModelLoggingInterface = {
+    debug: (msg: string) => { console.log(msg) },
+    info: (msg: string) => { console.log(msg) },
+    warn: (msg: string) => { console.log(msg) },
+    error: (msg: string) => { console.log(msg) }
+};
+Logger.setLogger(mockedLogger);
+//const generator = new TypeScriptGenerator({ modelType: 'interface' });
+//const interfaceModel = await generator.generate(...);
```
Comment (magicmatatjahu) [inline docs/logging.md:23]: Maybe add some comment to these lines or remove them :)

Comment (jonaslagoni) [reply docs/logging.md]: Added an extra comment to clarify it.

File: docs/logging.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const customLogger: ModelLoggingInterface = {
+    debug: (msg: string) => { console.log(msg) },
+    info: (msg: string) => { console.log(msg) },
+    warn: (msg: string) => { console.log(msg) },
+    error: (msg: string) => { console.log(msg) }
+};
+Logger.setLogger(mockedLogger);
+// Now use the library as normal without doing anything else. 
+//  const generator = new TypeScriptGenerator({ modelType: 'interface' });
+//  const interfaceModel = await generator.generate(...);
```
Comment (magicmatatjahu) [inline docs/logging.md:24]:
```suggestion
Logger.setLogger(mockedLogger);

// Now use the library as normal without doing anything else. 
// const generator = new TypeScriptGenerator({ modelType: 'interface' });
// const interfaceModel = await generator.generate(...);
```

**Review by magicmatatjahu — APPROVED** [review]
LGTM! Only one improvement :)

**Review by magicmatatjahu — APPROVED** [review]

**Comment by jonaslagoni** [comment]
Thanks 👍

**Pull request merged**